CSVSource/src/test/java/io/vantiq/extsrc/CSVSource/TestCSVReaderFixedLength.java

```diff
@@ -0,0 +1,168 @@
+        AppendFileForTest(testFullFilePath, "0000000000009 itemName123412341234 1 100000 200000 01");
+        AppendFileForTest(testFullFilePath, "0000000000010 itemName123412341234 1 100000 200000 01");
+        AppendFileForTest(testFullFilePath, "0000000000011 itemName123412341234 1 100000 200000 01");
+        AppendFileForTest(testFullFilePath, "0000000000012 itemName123412341234 1 100000 200000 01");
+        AppendFileForTest(testFullFilePath, "0000000000013 itemName123412341234 1 100000 200000 01");
+
+        config.put("maxLinesInEvent", 3);
+        try {
+            ArrayList<Map<String, String>> lastSegment = CSVReader.executeFixedRecord(testFullFilePath, config, null);
+            assertTrue("Unexpected number of segments", CSVReader.segmentList.size() == 3);
+
+            assertTrue("Unexpected lines of csv file in array", lastSegment.size() == 2);
+            assertTrue("Unexpected values in first of line of csv file in array", lastSegment.get(0).size() == 6);
+
+            String v = lastSegment.get(0).get("code");
+            assertTrue("Unexpected field code in first of line of csv file in array", v.equals("0000000000012"));
+            v = lastSegment.get(0).get("name");
+            assertTrue("Unexpected field name in first of line of csv file in array",
+                    lastSegment.get(0).get("name").equals("432143214321emaNmeti"));
+            assertTrue("Unexpected field price in first of line of csv file in array",
+                    lastSegment.get(0).get("price").equals("100000"));
+
+            java.util.LinkedHashMap firstSegment = (java.util.LinkedHashMap) CSVReader.segmentList.get(0);
+            ArrayList<Map<String, String>> firstSegmentLines = (ArrayList<Map<String, String>>) firstSegment
+                    .get("lines");
+
+            assertTrue("Unexpected lines of csv file in array", firstSegment.size() == 3);
+            assertTrue("Unexpected values in first of line of csv file in array", firstSegmentLines.size() == 3);
+
+            String v1 = firstSegmentLines.get(0).get("code");
+            assertTrue("Unexpected field code in first of line of csv file in array", v1.equals("0000000000006"));
+            v = firstSegmentLines.get(0).get("name");
+            assertTrue("Unexpected field name in first of line of csv file in array",
+                    firstSegmentLines.get(0).get("name").equals("432143214321emaNmeti"));
+            assertTrue("Unexpected field price in first of line of csv file in array",
+                    firstSegmentLines.get(0).get("price").equals("100000"));
+        } catch (InterruptedException ex) {
+            assertTrue("Exception " + ex, false);
+        }
+    }
+
+    @Test
+    public void testReadSimpleFile2Record() {
+        CreateFileForTest(testFullFilePath, "0000000000006 itemName123412341234 1 100000 200000 01");
+        AppendFileForTest(testFullFilePath, "0000000000007 itemName123412341234 1 100000 200000 01");
```

Comment:
Suppose you add 

```
        AppendFileForTest(testFullFilePath, "0000000000008 itemName123412341234 1 100000 200000 01junkjunkjunk");
```
I think it fails.  I think we need to handle this some way, don't we?

Reply:
the 'junkjunkjunk' will be the beginning of the next record , i don't think i can help here